scripts/retrieve_ci_failures.py

```diff
@@ -0,0 +1,307 @@
+        }
+
+    r = requests.get(
+        "https://sql.telemetry.mozilla.org/api/queries/111789/results.json?api_key={}".format(
+            utils.get_secret("REDASH_API_KEY")
+        )
+    )
+    fixed_by_commit_elements = r.json()["query_result"]["data"]["rows"]
+
+    fixed_by_commit_elements = [
+        element
+        for element in fixed_by_commit_elements
+        if element["repository_name"] == "autoland"
+    ]
+
+    for element in fixed_by_commit_elements:
+        bug_id = int(element["bug_id"])
+
+        if bug_id not in fixed_by_commit_pushes:
+            fixed_by_commit_pushes[bug_id] = {
+                "failures": [],
+                "commits": [],
+            }
+
+        fixed_by_commit_pushes[bug_id]["failures"].append(
+            {
+                "task_name": element["job_type_name"],
+                "task_id": element["task_id"],
+                "retry_id": element["retry_id"],
+                "failure_lines": element["failure_lines"],
+            }
+        )
+
+    print(f"Analyzing {len(fixed_by_commit_pushes)} 'fixed by commit' pushes.")
+
+    backouts_by_bug_id = defaultdict(int)
+    for commit in repository.get_commits(include_backouts=True):
+        if commit["bug_id"] not in fixed_by_commit_pushes:
+            continue
+
+        fixed_by_commit_pushes[commit["bug_id"]]["commits"].append(commit)
+
+        if commit["backsout"]:
+            backouts_by_bug_id[commit["bug_id"]] += 1
+
+    # Skip cases where there is no relanding.
+    no_relanding_bugs = set()
+    for bug_id, obj in fixed_by_commit_pushes.items():
+        if not any(not commit["backedoutby"] for commit in obj["commits"]):
+            no_relanding_bugs.add(bug_id)
+
+    print(f"{len(no_relanding_bugs)} cases removed because there was no relanding.")
+
+    for bug_id in no_relanding_bugs:
+        del fixed_by_commit_pushes[bug_id]
+
+    print(f"{len(fixed_by_commit_pushes)} 'fixed by commit' pushes left to analyze.")
+
+    # Skip cases where there are multiple backouts associated to the same bug ID.
+    multiple_backouts = set()
+    for bug_id, backouts in backouts_by_bug_id.items():
+        if backouts > 1:
+            if bug_id in fixed_by_commit_pushes:
+                multiple_backouts.add(bug_id)
+
+    print(
+        f"{len(multiple_backouts)} cases removed because there were multiple backouts in the same bug."
+    )
+
+    for multiple_backout in multiple_backouts:
+        del fixed_by_commit_pushes[multiple_backout]
+
+    print(f"{len(fixed_by_commit_pushes)} 'fixed by commit' pushes left to analyze.")
+
+    # Skip cases where there is no backout (and so the fix was a bustage fix).
+    no_backouts = set()
+    for bug_id in fixed_by_commit_pushes.keys():
+        if bug_id not in backouts_by_bug_id:
+            print(f"No backouts on {bug_id}")
+            no_backouts.add(bug_id)
+
+    print(
+        f"{len(no_backouts)} cases removed because there were no backouts in the bug."
+    )
```

Comment:
We have not removed them yet; we could move them after removing them or tweak the wording to reflect that.